modules/opensearch/includes/opensearch.search.inc

```diff
@@ -147,7 +183,24 @@ function opensearch_search_search(TingSearchRequest $ting_query) {
   // Whether to return all materials from matched collections.
   $options['allObjects'] = $ting_query->getPopulateCollections();
 
-  return new OpenSearchTingSearchResult(ting_do_search($query, $ting_query->getPage(), $ting_query->getCount(), $options), $ting_query);
+  $result = new OpenSearchTingSearchResult(ting_do_search($query, $ting_query->getPage(), $ting_query->getCount(), $options), $ting_query);
+
+  // Handle the situation where opensearch sets a default sort.
+  if (empty($result->getSearchRequest()->getSorts()) && !empty($result->openSearchResult->sortUsed)) {
+    $sort_used = $result->openSearchResult->sortUsed;
+    if ($sort_used === 'rank_main_title') {
+      $sort_used = 'rank_title';
+    }
+
+    // If the used field is a field we already support, pretend it was picked
+    // by the user all along.
+    if (in_array($sort_used, ['rank_title', 'rank_subject', 'rank_creator'])) {
```

Comment:
Should these keys come from somewhere instead of just having them as strings here?

I notice that they are also keys from `opensearch_search_sort_options()`.

Reply:
good point!